src/main/java/ch/njol/skript/expressions/ExprSets.java

```diff
@@ -47,60 +50,60 @@
 /**
  * @author Peter Güttinger
  */
-@Name("Items")
-@Description("Items or blocks of a specific type, useful for looping.")
+@Name("Sets")
+@Description("Collection sets of items or blocks of a specific type or colours, useful for looping.")
 @Examples({"loop items of type ore and log:",
 		"	block contains loop-item",
 		"	message \"Theres at least one %loop-item% in this block\"",
 		"drop all blocks at the player # drops one of every block at the player"})
-@Since("<i>unknown</i> (before 2.1)")
-public class ExprItems extends SimpleExpression<ItemStack> {
-	
+@Since("<i>unknown</i> (before 1.4.2), INSERT VERSION (colors)")
+public class ExprSets extends SimpleExpression<Object> {
+
 	static {
-		Skript.registerExpression(ExprItems.class, ItemStack.class, ExpressionType.COMBINED,
+		Skript.registerExpression(ExprSets.class, Object.class, ExpressionType.COMBINED,
 				"[(all [[of] the]|the|every)] item(s|[ ]types)", "[(all [[of] the]|the)] items of type[s] %itemtypes%",
-				"[(all [[of] the]|the|every)] block(s|[ ]types)", "[(all [[of] the]|the)] blocks of type[s] %itemtypes%");
+				"[(all [[of] the]|the|every)] block(s|[ ]types)", "[(all [[of] the]|the)] blocks of type[s] %itemtypes%",
+				"([all] [[of] the] colo[u]rs|(the|every) colo[u]r)");
```

Comment:
```suggestion
				"[((all [[of] the]|the) colo[u]rs|every colo[u]r)]");
```
Fixes allowing patterns like `of the colors`

Reply:
You moved the optional too far, the whole syntax is optional, I get what you mean though.